src/Jobs/getUsersData.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace BristolSU\UnionCloud\Jobs;
+
+use BristolSU\Support\User\User;
+use GuzzleHttp\Exception\ClientException;
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Queue\SerializesModels;
+use BristolSU\UnionCloud\UnionCloud\UnionCloud;
+use Illuminate\Support\Facades\Log;
+
+class GetUsersData implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    protected int $page;
+    protected int $pageCount;
+    protected int $delayBy;
+    protected array $users;
+    protected $repository;
+
+    protected int $requestLimit = 1;
+
+    /**
+     * Create a new job instance.
+     *
+     * @param int $page
+     * @param int $pageCount
+     * @param int $delayBy
+     * @param array $users
+     */
+    public function __construct(int $page, int $pageCount, int $delayBy, array $users = [])
+    {
+        $this->page = $page;
+        $this->pageCount = $pageCount;
+        $this->delayBy = $delayBy;
+        $this->users = $users;
+
+        $this->requestLimit = config('unioncloud-portal.users_per_minute');
+
+        // Init Repository
+        $this->repository = app(UnionCloud::class);
```

Comment:
Running in a queue, I don't think this will work? It'll serialize all the properties on the job, try and serialise the repository but fail (I think). Typehinting the repository on the `handle` method should fix it

Reply:
Should also be a local variable to avoid job serialization issues